server/core/src/main/java/org/infinispan/server/core/LifecycleCallbacks.java

```diff
@@ -40,22 +43,23 @@ public void cacheManagerStarting(GlobalComponentRegistry gcr, GlobalConfiguratio
       cacheRegistry.registerInternalCache(SERVER_STATE_CACHE, getServerStateCacheConfig(globalConfiguration).build(),
             EnumSet.of(InternalCacheRegistry.Flag.PERSISTENT));
 
+      SerializationContext serCtx = gcr.getComponent(PersistenceMarshallerImpl.class, KnownComponentNames.PERSISTENCE_MARSHALLER).getSerializationContext();
```

Comment:
I have to admit I am not a fan of referencing `PersistenceMarshallerImpl` everywhere.

The `PersistenceMarshaller` interface is already intrinsically bound to protostream through the `register` method it doesn't seem far fetched for it to also be able to return the `SerializationContext` instance.

But this can be fixed in a different PR.

Reply:
That's a good point. I'll make a note of it and add a commit in a follow up PR. I don't think it's worth waiting another CI run for